### PR TITLE
add standardized tests for flags in backup cmds

### PR DESCRIPTION
**Pull request by ryanfkeepers**
#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :robot: Supportability/Tests

#### Test Plan

- [x] :zap: Unit test

### CODE REVIEW

File: src/cli/backup/backup.go
Comment (meain) [inline src/cli/backup/backup.go:282]: I'm not sure if this is the best approach. Maybe we should split adding flags and the use of `utils.AddCommand`?

Comment (ryanfkeepers) [reply src/cli/backup/backup.go:282]: It's not the best approach, agreed.  But it's also the pattern we've established in other areas of code.  Let's move this PR forward and we can address the pattern separately.

**Comment by ryanfkeepers** [comment]
@meain could this get a final review, please?

**Comment by pandeyabs** [comment]
I'll also review it shortly today. Sorry about the delay!

**Review by meain — APPROVED** [review]
Woops, I somehow missed the notification on this one.

File: src/cli/backup/onedrive_test.go
Comment (pandeyabs) [reply src/cli/backup/onedrive_test.go:224]: Same goes for assertions

File: src/cli/backup/backup.go
Comment (pandeyabs) [reply src/cli/backup/backup.go:282]: Can we create an issue to track this please? As an alternative, I feel we could add some lightweight integration tests instead, on an empty mailbox/drive/sp/groups.

**Review by pandeyabs — APPROVED** [review]
Approving to unblock. Not sure if we want to proliferate use of `isTestMode()` pattern in code.

File: src/cli/backup/backup.go
Comment (meain) [reply src/cli/backup/backup.go:282]: I don't know if we should be adding integrations test for these, as that feels like adding unnecessary complexity just to test availability of flags.

Comment (ryanfkeepers) [reply src/cli/backup/backup.go:282]: Agreed, we should avoid integration tests for this purpose.

File: src/cli/backup/onedrive_test.go
Comment (ryanfkeepers) [reply src/cli/backup/onedrive_test.go:224]: I have follow-up PRs which already move a large set of common assertions to shared code.

**Pull request merged**
